### PR TITLE
W-10952349: Review GitHub documentation

### PR DESCRIPTION
**Pull request by mlischetti**

### CODE REVIEW

**Comment by uip-robot-zz** [comment]
Git2Gus App is installed but the `.git2gus/config.json` doesn't exist.

File: Readme.md [hunk 1]
```diff
@@ -1,32 +1,129 @@
...
+
+## Grimoire
+
+A `grimoire` is a collection of `spells` from a given `wizard`. 
+
+## Spells
+
+A `spell` is a very simple way to distribute and share a DataWeave transformation.
+
+A `spell` is just executables scripts that can be called from the command-line using the `spell` name.
```
Comment (afelisatti) [inline Readme.md]:
```suggestion
A `spell` is just an executable script that can be called from the command line using the `spell`'s name.
```

File: Readme.md [hunk 2]
```diff
@@ -1,32 +1,129 @@
...
+
+## DataWeave Grimoire
+
+DataWeave Grimoire is the default grimoire provided by DataWeave CLI.
```
Comment (afelisatti) [inline Readme.md]:
```suggestion
The DataWeave Grimoire is the default grimoire provided by the DataWeave CLI.
```

File: Readme.md [hunk 3]
```diff
@@ -1,32 +1,129 @@
...
+
+In order to add a new `wizard` use the DataWeave CLI
+
+```bash
+dw --add-wizard <wizard_name>
 ```
-  %dw 2.0
-  ---
-  "Hi Wise Wizard" 
+The wizard grimoire is going to be cloned at `{user.home}/.dw/grimoires`
```
Comment (afelisatti) [inline Readme.md]:
```suggestion
The wizard's grimoire is going to be cloned at `{user.home}/.dw/grimoires`
```

File: Readme.md [hunk 4]
```diff
@@ -1,32 +1,129 @@
...
+
+:warning: **Do NOT add a wizard you don't trust**
```
Comment (afelisatti) [inline Readme.md]:
```suggestion
:warning: **Do NOT add wizards you don't trust**
```

File: Readme.md [hunk 5]
```diff
@@ -1,32 +1,129 @@
...
-## How to test Spells locally before pushing to a repo
+### Running a Spell From the DataWeave Grimoire
 
-Just run your dwl script with:
+```bash
+dw --spell <spell_name>
 ```
-dw -f <path_to_main_script>
+
+### Running a Spell From a Grimoire a Local Folder
```
Comment (afelisatti) [inline Readme.md]:
```suggestion
### Running a Spell From a Local Grimoire
```

Comment (afelisatti) [reply Readme.md]: I'm not sure if this was the meaning. Does it mean a grimoire you have somewhere besides `.dw/grimoires`?

File: Readme.md [hunk 6]
```diff
@@ -1,32 +1,129 @@
...
+
+```bash
+dw --local-spell <spell_name>
 ```
+
+## Becoming a DataWeave Wizard
+
+### Step 1: Create the GitHub Repository
```
Comment (afelisatti) [inline Readme.md]:
```suggestion
### Step 1: Create Your GitHub Repository
```

File: Readme.md [hunk 7]
```diff
@@ -1,32 +1,129 @@
...
+
+Inside your cloned repository run
+
+```bash
+dw --new-spell <spell_name>
+```
+
+### Step 3: Edit Your Spell
+
+Use [DataWeave VSCode Plugin](https://marketplace.visualstudio.com/items?itemName=MuleSoftInc.dataweave) to develop your script.
```
Comment (afelisatti) [inline Readme.md:111]: Is there a way to launch VS Code from the terminal?

**Review by afelisatti — CHANGES_REQUESTED** [review]

File: Readme.md (hunk 7 above)
Comment (mlischetti) [reply Readme.md:111]: Yes, I added the command to do it.

File: Readme.md (hunk 5 above)
Comment (mlischetti) [reply Readme.md]: This command runs the spell based on your current path. I added a few more references.

File: Readme.md [hunk 8]
```diff
@@ -1,32 +1,138 @@
...
 
-You can create your own grimore and be part of the DataWeave Wizards Secret Club.
+You can create your own grimoire and be part of the DataWeave Wizards Secret Club.
 
-## How to create a SPELL
+Let's start defining some concepts:
 
-- wizard-name = Github User Name
+## Wizard
+A `wizard` is just a name that we trust and we want to have all their `spells` available.
```
Comment (machaval) [inline Readme.md]: I would change name for user.

File: Readme.md [hunk 9]
```diff
@@ -1,32 +1,138 @@
...
- - And then write your script inside Main.dwl for example:
+A `wizard` has a `grimoire`, a GitHub repository that is inferred by `https://github.com/${wizard_name}/${wizard_name}-data-weave-grimoire`, that includes all the `spells` from this `wizard`.
+
+## Grimoire
+
+A `grimoire` is a collection of `spells` from a given `wizard`. 
+
+## Spells
+
+A `spell` is a very simple way to distribute and share a DataWeave transformation.
```
Comment (machaval) [inline Readme.md]: I would change the spells docs with something like this. 

A `spell` is a DataWeave application that contains one or more DataWeave Transformations (entry points) and is hosted in a GitHub repo inside a grimoire. This makes it very easy to share and distribute across other `wizards` (users).

File: Readme.md [hunk 10]
```diff
@@ -1,32 +1,138 @@
...
-All you need is 
- - A repo `*<wizard-name>*-data-weave-grimoire`
- - A folder that is going to be your spell name i.e HelloWorld/
- - Then a file Main.dwl that needs to be under <spell_name>/src/Main.dwl in our case HelloWorld/src/Main.dwl
- - And then write your script inside Main.dwl for example:
+A `wizard` has a `grimoire`, a GitHub repository that is inferred by `https://github.com/${wizard_name}/${wizard_name}-data-weave-grimoire`, that includes all the `spells` from this `wizard`.
+
+## Grimoire
+
+A `grimoire` is a collection of `spells` from a given `wizard`.
```
Comment (machaval) [inline Readme.md]: A `Grimoire` is a collection of `spell` created by a `wizard`. The grimoire is hosted in a GitHub repo in the default branch. Currently there is no support for versioning or branching scheme.

File: Playground/Readme.md [hunk 11]
```diff
@@ -1,13 +1,8 @@
 # Playground Spell
+This spell will launch the playground UI. 
+For the server is uses our [data-weave-io](https://github.com/mulesoft-labs/data-weave-io) module. This module is still a POC but is fully functional right now.
```
Comment (afelisatti) [inline Playground/Readme.md]:
```suggestion
For the server it uses our [data-weave-io](https://github.com/mulesoft-labs/data-weave-io) module. This module is still a POC but is fully functional right now.
```

**Review by afelisatti — CHANGES_REQUESTED** [review]

**Review by afelisatti — APPROVED** [review]

**Pull request merged**